spec/models/notification_spec.rb

```diff
@@ -371,4 +371,20 @@
       end.to change(Notification, :count).by(1)
     end
   end
+
+  describe "#remove_each_without_delay" do
+    let(:mention) { create(:mention, user_id: user.id, mentionable_id: comment.id, mentionable_type: "Comment") }
+    let(:comment) { create(:comment, user_id: user.id, commentable_id: article.id) }
+    let(:notifiable_collection) { [mention] }
+
+    before do
+      Notification.send_mention_notification_without_delay(mention)
+    end
+
+    it "removes each mention related notifiable" do
+      expect do
+        Notification.remove_each_without_delay(notifiable_collection, nil)
```

Comment:
I don't recommend using `_without_delay` methods in specs because eventually, we will want to get rid of them not to clutter the codebase. I suggest using `perform_enqueued_jobs` helper and call the regular `remove_each` method instead.

Reply:
I have removed the send_mention without delay and used notification + mention creation instead.